### PR TITLE
Fixed few misra + serious missing parenthesis in function call

### PR DESCRIPTION
**Pull request by Hadatko**
Signed-off-by: Cervenka Dusan <cervenka@acrios.com>

### CODE REVIEW

File: lib/virtio/virtqueue.c [hunk 1]
```diff
@@ -63,11 +63,11 @@ int32_t virtqueue_create(uint16_t id,
 {
     struct virtqueue *vq    = VQ_NULL;
     volatile int32_t status = VQUEUE_SUCCESS;
-    uint32_t vq_size        = 0;
+    uint32_t vq_size;
```
Comment (MichalPrincNXP) [inline lib/virtio/virtqueue.c:66]: Why removing the vq_size initial value? I would rather changed it to = 0U;

Comment (Hadatko) [reply lib/virtio/virtqueue.c:66]: The value is initialized with first usage of it `vq_size = sizeof(struct virtqueue);`
That is why misra is complaining. So it depends how much you want implement misra checks.

File: lib/include/virtqueue.h [hunk 2]
```diff
@@ -41,16 +41,16 @@ typedef uint8_t boolean;
...
-#define ERROR_INVLD_DESC_IDX     (VQ_ERROR_BASE - 2)
-#define ERROR_EMPTY_RING         (VQ_ERROR_BASE - 3)
-#define ERROR_NO_MEM             (VQ_ERROR_BASE - 4)
-#define ERROR_VRING_MAX_DESC     (VQ_ERROR_BASE - 5)
-#define ERROR_VRING_ALIGN        (VQ_ERROR_BASE - 6)
-#define ERROR_VRING_NO_BUFF      (VQ_ERROR_BASE - 7)
-#define ERROR_VQUEUE_INVLD_PARAM (VQ_ERROR_BASE - 8)
-
-#define VQUEUE_SUCCESS (0)
+#define ERROR_VRING_FULL         (VQ_ERROR_BASE - 1U)
```
Comment (MichalPrincNXP) [inline lib/include/virtqueue.h:44]: VQ_ERROR_BASE is signed so why it is needed to change the 1 to 1U? What MISRA rule is broken here?

File: lib/virtio/virtqueue.c (hunk 1 above)
Comment (MichalPrincNXP) [reply lib/virtio/virtqueue.c:66]: I guess when using 0U initial value, the Misra should be OK. What Misra rule this relates to?

Comment (Hadatko) [reply lib/virtio/virtqueue.c:66]: With `uint32_t vq_size =0U;` still "Variable 'vq_size' is assigned a value that is never used."
But sorry this one is not misra issue but cppcheck issue which i am also using. So you can ignore it if you want.

File: lib/include/virtqueue.h (hunk 2 above)
Comment (Hadatko) [reply lib/include/virtqueue.h:44]: Hmm this looks like wrong identification. Sorry. will revert

File: lib/virtio/virtqueue.c (hunk 1 above)
Comment (MichalPrincNXP) [reply lib/virtio/virtqueue.c:66]: OK, I would rather stay with current code in this case, our Misra check does not indicate any issue.

File: lib/include/virtqueue.h (hunk 2 above)
Comment (MichalPrincNXP) [reply lib/include/virtqueue.h:44]: ok, thanks

File: lib/virtio/virtqueue.c (hunk 1 above)
Comment (Hadatko) [reply lib/virtio/virtqueue.c:66]: Ok i will add U as you suggested. Misra is likely not complaining because value is not used...

**Review by MichalPrincNXP — APPROVED** [review]

**Comment by MichalPrincNXP** [comment]
Thank you!

**Comment by Hadatko** [comment]
You are welcome. Sorry for few wrong changes.

**Pull request merged**